api_server/src/http_service.rs

```diff
@@ -806,7 +807,10 @@ mod tests {
         assert!(checked_id("dummy").is_ok());
         assert!(checked_id("dummy_1").is_ok());
         assert!(checked_id("") == Err(Error::EmptyID));
-        assert!(checked_id("dummy!!") == Err(Error::NonAlphanumericID));
+        match checked_id("dummy!!").unwrap_err() {
```

Comment:
Why not an `assert!`?

Reply:
Unit tests were failing when I was using 
`assert!(checked_id("dummy!!") == Err(Error::NonAlphanumericID))`.
I don't have a great deal of rust experience so I'm not sure what went wrong. I tried adding `#[derive(PartialEq)]` to the error type but it does not work for `SerdeJson` variant.

Reply:
@aghecenco this is what I get from the logs when running `cargo test`
```
thread 'http_service::tests::test_checked_id' panicked at 'assertion failed: `(left == right)`
  left: `NonAlphanumericID`,
 right: `NonAlphanumericID`', api_server/src/http_service.rs:809:9
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

Reply:
You can `assert` on the string representation instead, since you derived `Debug`.
```rust
assert_eq!(format!("{:?}", checked_id("dummy!!")), <insert json error string here>);
```